### PR TITLE
HPCC-18688 HTPasswd security manager feature support incomplete

### PR DESCRIPTION
**Pull request by RussWhitehead**
When htpasswd security is enabled, the Pre-Flight check for
OPERATIONS -> CLUSTER PROCESSES does not have a submit button so unable to
do pre-flight check or also get Roxie Metrics. This PR adds a security
feature map to HTPasswd, granting access to all authenticated users.

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@wangkx Please review

File: system/security/plugins/htpasswdSecurity/htpasswdSecurity.cpp [hunk 1]
```diff
@@ -112,6 +112,47 @@ class CHtpasswdSecurityManager : public CBaseSecurityManager
 		return authmap;
 	}
 
+    IAuthMap * createFeatureMap(IPropertyTree * authconfig)
+    {
+	    CAuthMap* feature_authmap = new CAuthMap(this);
+
+	    IPropertyTreeIterator *feature_iter = NULL;
+	    feature_iter = authconfig->getElements(".//Feature");
```
Comment (wangkx) [inline system/security/plugins/htpasswdSecurity/htpasswdSecurity.cpp]: Why not use 'Owned<IPropertyTreeIterator> it = tree->getElements(".//Feature");'? You may use 'ForEach(*it)' after that. You will not have to check NULL and isValid.

File: system/security/plugins/htpasswdSecurity/htpasswdSecurity.cpp [hunk 2]
```diff
@@ -112,6 +112,47 @@ class CHtpasswdSecurityManager : public CBaseSecurityManager
...
+	        while(feature_iter->isValid())
+	        {
+	            feature = &feature_iter->query();
+	            if (feature)
+	            {
+	                StringBuffer pathstr, rstr, required, description;
+	                feature->getProp("@path", pathstr);
+	                feature->getProp("@resource", rstr);
+	                feature->getProp("@required", required);
+	                feature->getProp("@description", description);
```
Comment (wangkx) [inline system/security/plugins/htpasswdSecurity/htpasswdSecurity.cpp]: Could one of those 4 attributes be empty?

**Review by wangkx — COMMENTED** [review]
@RussWhitehead 2 minor comments.

**Comment by RussWhitehead** [comment]
@wangkx Please rereview

**Comment by wangkx** [comment]
@RussWhitehead looks fine now.

**Comment by RussWhitehead** [comment]
@richardkchapman Please merge in 6.4.6

**Pull request merged**